### DISCUSSION TITLE
CPUThrottlingHigh Prometheus notifiation

### DISCUSSION THREAD

**Discussion by giovanni-motterle-mia-care** [mia-platform/community#176] [Q&A] [answered]
Hi, I'm trying to understand why a particular pod is constantly notifing for CPU throttling even if the base CPU usage is really low compared to the CPU request.
Here is a screenshot of grafana.
![Screenshot from 2023-01-20 10-20-20](https://user-images.githubusercontent.com/110187011/213659955-ceb1e428-e3e2-4138-8712-92bca346232a.png)
What does **100%** of throttling means? And why is a predefines threshold of 25% always set?

**Comment by fredmaggiowski** [comment, marked as answer]
Hi, CPU for your service gets throttled when it is trying to consume more CPU than it's allowed to, based on the CPU limit you configured.

Consider that when seeing grafana dashboard from prometheus your are not seeing raw points collected but you'll see a media value calculated based on the sampling time, so your service may have very short but high CPU spikes that gets mitigated by the mathematical media and cannot be seen in the dashboard.

**Reply by giovanni-motterle-mia-care** [reply]
thanks, but what does it means that the CPU throttling is costantly around 60%? 60% of what? And why the 25% default threashold?

**Reply by fredmaggiowski** [reply]
There's no such thing as a "default threshold" the service is simply too greedy on CPU you are giving to it. 

Whether it is due to how the code or to an improper (too low) configuration of the CPU limit it's an information only you can assess 🤷.

**Reply by giovanni-motterle-mia-care** [reply]
Here is another example
![Screenshot from 2023-01-20 15-40-45](https://user-images.githubusercontent.com/110187011/213726347-27f238fb-9228-4c47-bed9-b7d104805df6.png)
Why is grafana displaying a line at 25%? And is this the threashold to send the notification?
Moreover, I understand the concept of spikes and means but in the first graph the throttling is **constantly** at ~60% and at the same time the mean usage is 1/4 of the request

**Reply by fredmaggiowski** [reply]
Yes, sorry I misunderstood, that line at 25% throttle shows you the threshold for the alerting system

> Moreover, I understand the concept of spikes and means but in the first graph the throttling is constantly at ~60% and at the same time the mean usage is 1/4 of the request

The throttle is not related to the configured CPU request but calculated based on the CPU limit and available CPUs in the node

**Reply by giovanni-motterle-mia-care** [reply]
ok, thanks !
